pallets/task/src/lib.rs

```diff
@@ -298,6 +302,11 @@ pub mod pallet {
 			// Check that the extrinsic was signed and get the signer.
 			let signer = ensure_signed(origin)?;
 
+			let task = Self::tasks(&task_id).ok_or(<Error<T>>::TaskNotExist)?;
+			// pay volunteer
+			let sub_account = Self::account_id(&task_id);
```

Comment:
We are creating a separate unique account based on taskID to serve as an intermediary for a balance transfer?

Can this account be controlled?

Reply:
So each task_id has unique sub_account from pallet, so it is secure (I feel this is same as PDAs in Solana). But if you found anything wrong with this I am okay for other options.

Reply:
So it is not exactly like PDAs in solana but to me it seems quite safe as it is based on Task::Hash which is unique (as guaranteed by Hashing algorithms) see how sub_account is created at here https://substrate.stackexchange.com/questions/477/how-can-we-get-a-pallets-sub-account-address-using-polkadotjs-api